generators/javascript/lists.js

```diff
@@ -299,6 +299,34 @@ Blockly.JavaScript['lists_getSublist'] = function(block) {
   return [code, Blockly.JavaScript.ORDER_FUNCTION_CALL];
 };
 
+Blockly.JavaScript['lists_sort'] = function(block) {
+  // Block for sorting a list.
+  var listCode = Blockly.JavaScript.valueToCode(
+      block, 'LIST',
+      Blockly.JavaScript.ORDER_FUNCTION_CALL) || '[]';
+  var direction = block.getFieldValue('DIRECTION') === '1' ? 1 : -1;
+  var type = block.getFieldValue('TYPE');
+  var getCompareFunctionName = Blockly.JavaScript.provideFunction_(
+          'lists_get_sort_compare',
+  ['function ' + Blockly.JavaScript.FUNCTION_NAME_PLACEHOLDER_ +
+    '(type, direction) {',
+      '  var compareFuncs = {',
+      '    "numeric":  function(a, b) { ',
+      '        return parseFloat(a) - parseFloat(b) }, ',
```

Comment:
Semicolons needed at the end of these three return statements.  Also extra space can be trimmed off this string (and a couple other places in this generator).